components/rendering/src/shortcode.rs

```diff
@@ -111,12 +111,22 @@ fn render_shortcode(
         tera_context.insert("body", b.trim_right());
     }
     tera_context.extend(context.tera_context.clone());
-    let tpl_name = format!("shortcodes/{}.html", name);
 
-    let res = context
-        .tera
-        .render(&tpl_name, &tera_context)
-        .map_err(|e| Error::chain(format!("Failed to render {} shortcode", name), e))?;
+    let res = {
+        let tpl_name = format!("shortcodes/{}.html", name);
+        let default_tpl_name = format!("__zola_builtins/shortcodes/{}.html", name);
+        if context.tera.templates.contains_key(&tpl_name) {
+            context
+                .tera
+                .render(&tpl_name, &tera_context)
+                .map_err(|e| Error::chain(format!("Failed to render {} shortcode", name), e))?
+        } else {
+            context
+                .tera
+                .render(&default_tpl_name, &tera_context)
+                .map_err(|e| Error::chain(format!("Failed to render {} shortcode", name), e))?
```

Comment:
is it possible to call the render_template in utils instead of duplicating the logic here?

Reply:
Done.